File: PrefixSum/RangeSum/sum.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    vector<int> querySum(vector<int>& arr, vector<vector<int>>& queries) {
        // code here
        int n = arr.size();
        for(int i=1;i<n;i++){
            arr[i]+=arr[i-1];
        }
        int q = queries.size();
        vector<int>ans;
        for(int i=0;i<q;i++){
            int left = queries[i][0]-1;
            int right = queries[i][1]-1;
            if(left==0)ans.push_back(arr[right]);
            else ans.push_back(arr[right]-arr[left-1]);
        }
        return ans;
    }
};

File: PrefixSum/RangeSum/sum.py
class Solution:
    def querySum(self, arr, queries):
        n = len(arr)

        for i in range(1, n):
            arr[i] += arr[i - 1]

        ans = []

        for i in range(len(queries)):
            left = queries[i][0] - 1
            right = queries[i][1] - 1

            if left == 0:
                ans.append(arr[right])
            else:
                ans.append(arr[right] - arr[left - 1])

        return ans